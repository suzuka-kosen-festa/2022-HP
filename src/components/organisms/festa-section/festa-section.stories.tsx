import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import type { ComponentPropsWithoutRef } from "react";
import { FestaSection } from ".";

type T = typeof FestaSection;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  children:
    "第57回鈴鹿高専祭、テーマは「Re:ROAD」\n今回のテーマ「Re:ROAD」には「道（Road）」を創り直す（Re）という意味と、「弾をこめる・再補填する（Reload）」という意味を使い、「今年の準備をする」「来年以降、これから先のための高専祭を行う」という想いが込められています。\n\n2019年を最後に鈴鹿高専祭は行われていません。\nそのため過去のデータもあまり残っておらず本当に1から、いや0からのスタートです。\nまだ何色にも色づいていない第57回鈴鹿高専祭を今から色づけていきます。",
  subtitle: "今年のテーマ「Re:ROAD」について",
  title: "About",
};

export default {
  args,
  argTypes: {
    children: { description: "Value of main sentence in section", control: { type: "text" } },
    title: { description: "Title of section", control: { type: "text" } },
    subtitle: { description: "Sub title of section", control: { type: "text" } },
  },
  component: FestaSection,
} as Meta;

export const Default: Story = {};
