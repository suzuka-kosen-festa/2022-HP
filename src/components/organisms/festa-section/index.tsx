import type { FC } from "react";
import tw from "twin.macro";
import { Link } from "wouter";
import type { FestaSectionProperties } from "./types/model";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";
import { SubHeading } from "../../atoms/sub-heading";
import { Button } from "../../atoms/button";

const FestaSectionContainer = tw.section`flex flex-col justify-center items-center`;

const FestaSection: FC<FestaSectionProperties> = ({ buttonText, children, title, subtitle, ...rest }) => (
  <FestaSectionContainer aria-label="今年のテーマ「Re:ROAD」について" {...rest}>
    <Heading css={tw`mb-4 sm:mb-8`} colorKey="red">
      {title}
    </Heading>
    <SubHeading css={tw`mb-2.5 sm:mb-4`}>{subtitle}</SubHeading>
    <Text css={tw`whitespace-pre-wrap mb-6 sm:mb-16`}>{children}</Text>
    <Link to="/member">
      <Button colorKey="red" as="a" role="button">
        {buttonText}
      </Button>
    </Link>
  </FestaSectionContainer>
);

export { FestaSection };
