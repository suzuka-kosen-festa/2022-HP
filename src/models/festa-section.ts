import type { ComponentProps, ReactHTML } from "react";

export type FestaSectionProperties = Omit<ComponentProps<ReactHTML["section"]>, "children"> & {
  title: string;
  children: string;
};
