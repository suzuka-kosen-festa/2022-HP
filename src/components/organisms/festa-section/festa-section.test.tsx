import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./festa-section.stories";

const { Default } = composeStories(stories);

const options = {
  name: "今年のテーマ「Re:ROAD」について",
};

describe("(components) organisms/festa-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
