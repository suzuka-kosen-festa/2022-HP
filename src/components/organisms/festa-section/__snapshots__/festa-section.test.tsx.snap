// Vitest Snapshot v1

exports[`(components) organisms/festa-section > take snap shot 1`] = `
<div>
  <section
    aria-label="今年のテーマ「Re:ROAD」について"
    class="css-wkpw2c"
  >
    <h1
      class="css-1aaoobw-FestaSection"
    >
      About
    </h1>
    <h2
      class="css-gko5cy-FestaSection"
    >
      今年のテーマ「Re:ROAD」について
    </h2>
    <p
      class="css-19xv45d-FestaSection"
    >
      第57回鈴鹿高専祭、テーマは「Re:ROAD」
今回のテーマ「Re:ROAD」には「道（Road）」を創り直す（Re）という意味と、「弾をこめる・再補填する（Reload）」という意味を使い、「今年の準備をする」「来年以降、これから先のための高専祭を行う」という想いが込められています。

2019年を最後に鈴鹿高専祭は行われていません。
そのため過去のデータもあまり残っておらず本当に1から、いや0からのスタートです。
まだ何色にも色づいていない第57回鈴鹿高専祭を今から色づけていきます。
    </p>
  </section>
</div>
`;
